Automattic/pocket-casts-ios
# Transcripts: Refactor scroll to range textKit code

| 📘 Part of: #1848  |  
|:---:|

Fixes # 

This PR just simplifies up the TextKit code used to scroll to specific ranges.
In the process also removes a console warning when opening the Transcript view for the first time
```
UITextView 0x10f06b600 is switching to TextKit 1 compatibility mode because its layoutManager was accessed. Break on void _UITextViewEnablingCompatibilityMode(UITextView *__strong, BOOL) to debug.
```

## To test

1. Start the app
2. Ensure you the transcripts FF enabled
3. Open an episode with transcripts. Ex: any episode of Cautionary Tales
4. Tap on play
5. Open the full screen player
6. Tap on Transcript button
7. Scroll around in the transcript
8. Tap on Search
9. Search for a word in the transcript
10. Go up and down on search results, make sure you cycle from the first to last result and vice versa
11. Ensure that scroll works correctly
12. Show and hide the keyboard by tapping on the Transcript text and ensure the position in scroll is kept

## Checklist
